libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -1483,8 +1487,15 @@ public void onMediaUploadProgress(final String localMediaId, final float progres
     }
 
     @Override
-    public void onMediaUploadFailed(final String localMediaId) {
-        getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId));
+    public void onMediaUploadFailed(final String localMediaId, String errorType) {
+        switch (errorType) {
+            case "CONNECTION_ERROR":
```

Comment:
Using `CONNECTION_ERROR` is based off it being the only error type I observed while disabling my device's network connectivity with airplane mode. I'm uncertain if there are other error types that might be thrown.

Reply:
> I'm uncertain if there are other error types that might be thrown.

In my testing, I did not observe any other error type beyond `CONNECTION_ERROR`. I did note the existence of another similar generic MediaErrorType called [`NO_CONNECTION`](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/trunk/fluxc/src/main/java/org/wordpress/android/fluxc/store/MediaStore.java#L498).

Perhaps we should check for both:

```
switch (errorType) {
    case "CONNECTION_ERROR":
    case "NO_CONNECTION":
        getGutenbergContainerFragment().mediaFileUploadPaused(Integer.valueOf(localMediaId));
        break;
    default:
        getGutenbergContainerFragment().mediaFileUploadFailed(Integer.valueOf(localMediaId));
        break;
}
```

@ParaskP7 As the author of `NO_CONNECTION`, do you have any perspective on when `NO_CONNECTION` may be thrown instead of `CONNECTION_ERROR`? For some quick context, we are attempting to add a new  media upload state called "paused" when the device is offline (to invoke `mediaFileUploadPaused`, instead of marking the upload failed with `mediaFileUploadFailed`).

Reply:
Unless I misunderstand how failed media uploads work, we may not need to wait for an error to be thrown, and could instead listen to network status instead. I have outline a potential example in https://github.com/wordpress-mobile/WordPress-Android/pull/19884. If we went with this change, we would avoid this section of code entirely, and therefore the differences between `NO_CONNECTION` and `CONNECTION_ERROR` would not be relevant.

I would still be interested in learning the circumstances of when `NO_CONNECTION` can be thrown, and also learning if there is a need I'm not considering to wait for an explicit connection error to be thrown (instead of just listening to the connection status).